CliFx/Attributes/CommandArgumentAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+namespace CliFx.Attributes
+{
+    /// <summary>
+    /// Annotates a property that defines a command option.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property)]
+    public class CommandArgumentAttribute : Attribute
+    {
+        /// <summary>
+        /// The name of the argument, which is used in help text.
+        /// </summary>
+        public string? Name { get; set; }
+
+        /// <summary>
+        /// Whether the argument is required.
+        /// </summary>
+        public bool IsRequired { get; set; }
+
+        /// <summary>
+        /// Option description, which is used in help text.
+        /// </summary>
+        public string? Description { get; set; }
+
+        /// <summary>
+        /// The ordering of the argument. Lower values will appear before higher values. If no order is given, it will be placed last.
+        ///
+        /// <remarks>
+        /// If two arguments have the same value for <see cref="Order"/>, they will be ordered according to their <see cref="Name"/> properties, or the property name if <see cref="Name"/> is not set.
+        /// </remarks>
```

Comment:
Do you mean the order returned by `Type.GetProperties()`? If so, read this remark from the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.type.getproperties?view=netframework-4.8):

> The `GetProperties` method does not return properties in a particular order, such as alphabetical or declaration order. Your code must not depend on the order in which properties are returned, because that order varies.

Reply:
Do you think the properties should be ordered after `IsRequired` first and foremost? Required arguments must appear first anyways, so this would be an easy way to enforce it.

Reply:
In that case, should we make order a required parameter? I think I would want to make sure the order is always how I want it and not up to some internal algorithm.

Essentially, order to an argument is what name/shortname is to an option, it's a primary identifier.

Reply:
So a validation error if a consumer uses the same order twice?

Reply:
I would say yes.